clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -3574,6 +3574,39 @@ def DeleteArrayOp : CIR_Op<"delete.array">,
 // CallOp and TryCallOp
 //===----------------------------------------------------------------------===//
 
+def SE_All : I32EnumAttrCase<"All", 1, "all">;
+def SE_Pure : I32EnumAttrCase<"Pure", 2, "pure">;
+def SE_Const : I32EnumAttrCase<"Const", 3, "const">;
+
+def SideEffect : I32EnumAttr<
+    "SideEffect", "allowed side effects of a function",
```

Comment:
`SideEffect` -> `side_effect`, we don't use CamelCase for components that show up in IR.

Reply:
This is actually the C++ class name for the attribute, `I32EnumAttr` attributes do not have mnemonics.